File: Programmers/Lv_1/완주하지못한선수.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    string answer = "";

    sort(participant.begin(), participant.end());
    sort(completion.begin(), completion.end());

    for (auto iter = completion.rbegin(); iter != completion.rend(); ++iter) {
        if (participant.back() != *iter) {
            answer = participant.back();
            break;
        }
        participant.pop_back();
    }
    if (answer.empty()) answer = participant[0];

    return answer;
}


int main() {
    ios_base::sync_with_stdio(0);
    cin.tie(0);

    int pN, cN;
    string data;
    vector<string> participant, completion;

    cin >> pN;
    for (int i = 0; i < pN; i++) {
        cin >> data;
        participant.push_back(data);
    }
    cin >> cN;
    for (int i = 0; i < cN; i++) {
        cin >> data;
        completion.push_back(data);
    }

    cout << solution(participant, completion);

    return 0;
}

File: Programmers/Lv_1/완주하지못한선수.js
function solution(participant, completion) {
    participant.sort();
    completion.sort();

    for (let i = 0; i < participant.length; i++) {
        if (participant[i] !== completion[i]) {
            return participant[i];
        }
    }
}

function solution2(participant, completion) {
    let hashed = [];

    participant.forEach(entry => {
        hashed[entry] = hashed[entry] ? hashed[entry]+1 : 1;
    })

    completion.forEach(entry => {
        hashed[entry] = hashed[entry]-1;
    })

    for (let key in hashed) {
        if (hashed[key] >= 1) return key;
    }
}